nmostesting/suites/IS0401Test.py

```diff
@@ -1232,6 +1240,12 @@ def test_20(self, test):
                 node_devices = response.json()
                 for device in node_devices:
                     for control in device["controls"]:
+                        if self.is04_utils.compare_api_version(api["version"], "v1.3") >= 0 and \
+                                control["type"].startswith("urn:x-nmos:"):
+                            if ("authorization" in control and control["authorization"] != self.authorization) or \

```

Comment:
:exploding_head: https://github.com/AMWA-TV/nmos-discovery-registration/blob/v1.3.x/APIs/schemas/device.json#L71 Did not even know this was a thing

Is this a IS-10 requirement or BCP-003-2? I dont recall this being written anywhere :speak_no_evil:

Reply:
@prince-chrismc if a Node is using auth for its Node API, it ought* to be using it for its Connection API, Events API, etc. which are advertised with `urn:x-nmos:` control types.

*but if you're asking whether that 'ought' is explicitly written as SHOULD or MUST anywhere, that's a good question :-)
@andrewbonney?

Reply:
If it's only a SHOULD (or currently implicit), I suggest that this ought to result in `WARNING` here, but then the IS-05-02 and IS-08-02 test suites ought to `FAIL` if the IS-04 Device `control` doesn't match what they are testing.

Reply:
Yeah I'd be happy to switch that to a warning if desired, although I figure that if a real world deployment has authorization enabled for one API it would be an unusual choice to disable it elsewhere.

Reply:
> *but if you're asking whether that 'ought' is explicitly written as SHOULD or MUST anywhere, that's a good question :-)

Yes, your phrasing it much more clear than mine. I did not find it anywhere when I glazed over both drafts

Is it worth adding an issues/pr to the BCP (or IS) for this issue so the group can go over it in the next call?
it SHOULD but that does not make sense from a security point of view to allow, perhaps a MUST?

Reply:
> Yeah I'd be happy to switch that to a warning if desired, although I figure that if a real world deployment has authorization enabled for one API it would be an unusual choice to disable it elsewhere.

Agreed. But perhaps a WARNING is strong enough here _provided_ that this PR also adds a FAIL check to `IS0502Test.test_2` and refactors `IS0802.test_2` and `find_device_advertisement` slightly and does the same there?

Reply:
I'm not able to test it with my current setup, but the latest commit should address this.

Reply:
Another wrinkle here that @simonlo-sony has discovered is the check for `urn:x-nmos:` means the control `type` for [Manifest Base URL](https://github.com/AMWA-TV/nmos-parameter-registers/blob/master/device-control-types/manifest-base.md) still matches so the test is expecting `"authorization": true`. Using Manifest Base URL and specifying authorization would actually be a way to solve https://github.com/AMWA-TV/nmos-authorization-practice/issues/5 ??

Reply:
OK, @simonlo-sony and I are happy with this resolution, thanks, Andrew!
@prince-chrismc, look good to you?

Reply:
Yep, It's consistent, looks good